applications/functions/doc/README.md

```diff
@@ -0,0 +1,44 @@
+
+# Functions *serverless document processing*
+
+Functions enables dynamic call processing based on customized serverless function and callflow document as input.
+
+## Enabling Functions
+
+Functions has to be started manually by one of these method.
+
+In the case of manual loading, Erlang shell
+```shell
+functions_app:start().
+```
+Or `sup` command in Linux.
+```shell
+sup functions_app start
+```
+
+## Work around to enable *eval* on CouchDB
+
+Javascript's *new Function()*, which is similar to *eval*, is used to parse a piece of Javascript code in a string into a function object, that later to process a callflow document. It is considered more secure than *eval* since the *new Function()* code runs in a separate scope.
+
+However, both *eval* and *new Function()* are blocked by Content Security Policy in CouchDB. A workaround has to be applied to enable *eval*.
+
+### CouchDB 2.3
+Make sure *--eval* is explicitly added under query_servers of etc/local.ini.
+
+```
+[query_servers]
+COUCHDB_QUERY_SERVER_JAVASCRIPT="<absoluate path>/bin/couchjs --eval <absoluate path>/share/server/main.js"
+```
+
+If CouchDB is started by dev/run from the couchdb repo, please update dev/run with
+```
+qs_javascript = toposixpath("%s --eval %s" % (couchjs, mainjs))
+```
+
+### CouchDB 2.2 or earlier versions or BigCouch
+Make sure *--eval* is explicitly added under query_servers of etc/local.ini.
+
+```
+[query_servers]
+javascript = "<absoluate path>/bin/couchjs --eval <absoluate path>/share/server/main.js"
+```
```

Comment:
Just a note to add instructions for setting up alternative Couch connections in the system dataplan to isolate `functions` database operations from the main Couch cluster(s).